src/main/java/chatterchicken/tasklist/TaskList.java

```diff
@@ -0,0 +1,126 @@
+package chatterchicken.tasklist;
+
+import chatterchicken.CCException;
+import chatterchicken.task.Task;
+import chatterchicken.ui.Ui;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.concurrent.ForkJoinTask;
+
+/**
+ * The chatterchicken.tasklist.TaskList class represents a collection of tasks managed by the chatterchicken.ChatterChicken task manager.
+ * It allows for adding, marking, unmarking, and deleting tasks, as well as printing the list of tasks
+ * along with their respective indexes. Tasks can be of different types: ToDo, Deadline, and Event.
+ */
+public class TaskList implements Iterable<Task> {
+
+    private ArrayList<Task> taskList;
+
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Adds a new task to the task list based on the provided Task object.
+     * Displays a confirmation message with the added task's details.
+     *
+     * @param task The task to be added to the task list.
+     * @throws CCException If there is an error in adding the task.
+     */
+    public void addTask(Task task) throws CCException {
+        taskList.add(task);
+        System.out.println(ChatterChicken.LINE
+                + ChatterChicken.INDENT + "Got it. I've added this task:\n"
+                + ChatterChicken.INDENT_BIG + task.getTaskForPrinting() + "\n"
+                + ChatterChicken.INDENT + "Now you have " + taskList.size() + " tasks in the list."
+                + ChatterChicken.LINE);
+    }
+
+    /**
+     * Marks a task as done based on the provided input.
+     * Displays a confirmation message with the marked task's details.
+     *
+     * @param input The input containing task information to mark as done.
+     * @throws CCException If there is an error in marking the task or if the input is invalid.
+     */
+    public void markTask(String input) throws CCException {
+        try {
+            Task task = taskList.get(getIndex(input));
+            task.setDone(true);
+            System.out.println(ChatterChicken.LINE
+                    + ChatterChicken.INDENT + "Nice! I've marked this task as done:\n"
+                    + ChatterChicken.INDENT_BIG + task.getTaskForPrinting()
+                    + ChatterChicken.LINE);
```

Comment:
Perhaps delegating the printing to the UI will be better? 
I also suggest this for other methods in the TaskList.java.